Aplib.Tests/Desire/GoalTests.cs

```diff
@@ -112,6 +112,28 @@ public void Goal_WhereEvaluationIsPerformed_DoesNotInfluenceBelieveSet()
         beliefSetMock.Verify(beliefSetMock => beliefSetMock.UpdateBeliefs(), Times.Never);
     }
 
+    /// <summary>
+    /// Given a valid goal with heuristics
+    /// when the goal's heuristic function's result will be different in the next frame
+    /// the most recent heuristics are used
+    /// </summary>
+    [Fact]
+    public void Goal_WhereHeuristicsChange_UsesUpdatedHeuristics()
+    {
+        // Arrange
+        bool shouldSucceed = false;
+        Goal<IBeliefSet> goal = new TestGoalBuilder().WithHeuristicFunction(_ => shouldSucceed).Build();
+
+        // Act
+        CompletionStatus stateBefore = goal.GetStatus(It.IsAny<IBeliefSet>());
+        shouldSucceed = true; // Make heuristic function return a different value on next invoke
+        CompletionStatus stateAfter = goal.GetStatus(It.IsAny<IBeliefSet>());
+
+        // Assert
+        stateBefore.Should().Be(CompletionStatus.Unfinished);
+        stateAfter.Should().Be(CompletionStatus.Success);
```

Comment:
Write Assert without Moq methods.
```suggestion
        bool shouldSucceed = false;
        IBeliefSet beliefSet = Mock.Of<IBeliefSet>();
        Goal<IBeliefSet> goal = new TestGoalBuilder().WithHeuristicFunction(_ => shouldSucceed).Build();

        // Act
        CompletionStatus stateBefore = goal.GetStatus(beliefSet);
        shouldSucceed = true; // Make heuristic function return a different value on next invoke
        CompletionStatus stateAfter = goal.GetStatus(beliefSet);

        // Assert
        stateBefore.Should().Be(CompletionStatus.Unfinished);
        stateAfter.Should().Be(CompletionStatus.Success);
```

Reply:
Fixed!